src/targeting.js

```diff
@@ -181,6 +181,48 @@ export function newTargeting(auctionManager) {
     return [];
   };
 
+  /**
+   * Returns filtered ad server targeting for custom and allowed keys.
+   * @param {targetingArray} targeting
+   * @param {string[]} allowedKeys
+   * @return {targetingArray} filtered targeting
+   */
+  function getAllowedTargetingKeyValues(targeting, allowedKeys) {
+    logInfo(`Allowed default targeting keys [ ${allowedKeys.join(', ')}]`);
+    const defaultKeyring = Object.assign({}, CONSTANTS.TARGETING_KEYS, CONSTANTS.NATIVE_KEYS);
+    const defaultKeys = Object.keys(defaultKeyring);
+    targeting.map(adUnit => {
+      const adUnitCode = Object.keys(adUnit)[0];
+      const keyring = adUnit[adUnitCode];
+      const keys = keyring.filter(kvPair => {
+        const key = Object.keys(kvPair)[0];
+        // check if key is in default keys, if not, it's custom, we won't remove it.
+        const isCustom = defaultKeys.filter(defaultKey => key.indexOf(defaultKeyring[defaultKey]) === 0).length === 0;
+        // check if key explicitly allowed, if not, we'll remove it.
+        const found = isCustom || allowedKeys.find(allowedKey => {
+          const allowedKeyName = defaultKeyring[allowedKey];
+          // we're looking to see if the key exactly starts with one of our default keys.
+          // (which hopefully means it's not custom)
+          const found = key.indexOf(allowedKeyName) === 0;
+          return found;
+        });
+        const keyDisposition = found ? 'Keeping' : 'Removing';
+        const keyType = isCustom ? 'custom' : 'default';
+        logInfo(`${keyDisposition} ${keyType} targeting key ${key} for ${adUnitCode} adserverTargeting object`);
```

Comment:
Would it be possible to instead keep arrays of all the keys that we are removing and keeping and have one log for each that says these keys we've removed and these we've kept? This just seems like a lot of logs if we do it for each individual key. 

Or it might be better to have just the array of keys that will be allowed or one log of what we have removed.

Reply:
Makes sense, I'll collate into one log. Thanks!

Reply:
This has been updated.